spinn_front_end_common/interface/java_caller.py

```diff
@@ -0,0 +1,158 @@
+        else:
+            self._javaspinnaker_path = javaspinnaker_path
+        if not os.path.isdir(self._javaspinnaker_path):
+            raise ConfigurationException(
+                "No Java code found at {}".format(self._javaspinnaker_path))
+
+        self._machine_json = None
+        self._placement_json = None
+
+    def set_machine(self, machine):
+        """
+        Passes the machine in leaving this class to decide pass it to Java.
+
+        :param machine: A machine Object
+        """
+
+        algo = ConvertToJavaMachine()
+        path = os.path.join(self._json_folder, "machine.json")
+        self._machine_json = algo(machine, path)
+
+    def set_database(self, database_file):
+        """
+        Passes the database file in.
+
+        :param database_file: Path to the sql lite databse
+        :type database_file: str
+        """
+        self._database_file = database_file
+
+    def set_placements(self, placements, transceiver):
+        """
+        Passes in the placements leaving this class to decide pass it to Java.
+
+        This method may obtain extra information about he placements which is \
+            why it also needs the transceiver.
+
+        Currently the extra information extracted is recording region
+            base address but this could change if recording region saved in
+            the database.
+
+        Currently this method uses json but that may well change to using the
+            database.
+
+        :param placements: The Placements Object
+        :param transceiver: The Transceiver
+        """
+        path = os.path.join(self._json_folder, "java_placements.json")
+        self._placement_json = self._write_placements(
+            placements, transceiver, path)
+
+    def _write_placements(self, placements, transceiver, path):
+
+        # Read back the regions
+        json_obj = list()
+        for placement in placements:
+            json_placement = OrderedDict()
+            json_placement["x"] = placement.x
+            json_placement["y"] = placement.y
+            json_placement["p"] = placement.p
+            vertex = placement.vertex
+            json_vertex = OrderedDict()
+            json_vertex["label"] = vertex.label
+            json_vertex["recordedRegionIds"] = vertex.get_recorded_region_ids()
+            json_vertex["recordingRegionBaseAddress"] = \
+                vertex.get_recording_region_base_address(
+                    transceiver, placement)
+            json_placement["vertex"] = json_vertex
+            json_obj.append(json_placement)
+
+        # dump to json file
+        with open(path, "w") as f:
+            json.dump(json_obj, f)
+
+        return path
+
+    def get_all_data(self):
+        """
+        Gets all the data from the previously set placements
+        and put these in the previously set database.
+
+        """
+        jar_file = os.path.join(
+            self._javaspinnaker_path, "SpiNNaker-front-end",
```

Comment:
```suggestion
            self._java_spinnaker_path, "SpiNNaker-front-end",
```

Reply:
if changing that makes you happy..

Reply:
it does. it makes it read easier. and stops pycharm complaining about typos